pkg/workerapi/workerclient/workerclient.go

```diff
@@ -0,0 +1,51 @@
+// Package workerclient provides a way to communicate with a Wharf worker
+// server.
+package workerclient
+
+import (
+	"context"
+	"io"
+
+	v1 "github.com/iver-wharf/wharf-cmd/api/workerapi/v1"
+	"github.com/iver-wharf/wharf-core/pkg/logger"
+	"google.golang.org/grpc"
+)
+
+var log = logger.NewScoped("WORKER-CLIENT")
+
+// Client is an interface with methods to communicate with a Wharf worker server.
+type Client interface {
+	StreamLogs(ctx context.Context, req *LogsRequest, opts ...grpc.CallOption) (v1.Worker_StreamLogsClient, error)
+	StreamStatusEvents(ctx context.Context, req *StatusEventsRequest, opts ...grpc.CallOption) (v1.Worker_StreamStatusEventsClient, error)
+	StreamArtifactEvents(ctx context.Context, req *ArtifactEventsRequest, opts ...grpc.CallOption) (v1.Worker_StreamArtifactEventsClient, error)
+	DownloadArtifact(artifactID uint) (io.ReadCloser, error)
+
+	Close() error
+}
+
+// ClientOptions contains options that can be used in the creation
+// of a new client.
+type ClientOptions struct {
+	// InsecureSkipVerify disables cert verification if set to true.
+	//
+	// Should NOT be true in a production environment.
+	InsecureSkipVerify bool
+}
```

Comment:
nit: Double naming. The context is already client-specific. Suggest renaming to simply `Options`